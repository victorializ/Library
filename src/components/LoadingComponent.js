import React from 'react';
import { Loader } from 'semantic-ui-react';

import { ErrorMessage } from './UITable/ErrorMessage';
import { DefaultSuccessMessage } from './UITable/SuccessMessage';

function LoadingComponent({ 
    data, 
    error: { message },
    loading,
    children = DefaultSuccessMessage
}) {
    return (
        <>
            { message && <ErrorMessage text={message} /> }
            { data && children }
            { loading && 
                <div className='center'>
                    <Loader active inline='centered'/>
                </div>
            }
        </>
    );
}

export { LoadingComponent };