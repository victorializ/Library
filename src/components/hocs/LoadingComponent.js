import React from 'react';

import { Loader } from 'semantic-ui-react';
import { ErrorMessage } from '../UITable/ErrorMessage';

function LoadingComponent(props) {
    const { WrappedComponent, data, error: { message } } = props;
    
    return (
        data || message ? 
            message ? 
                <ErrorMessage text={message} /> 
                    :
                <WrappedComponent data={data} {...props} />  
                : 
            <div className='center'>
                <Loader active inline='centered'/>
            </div>
    )
}

export { LoadingComponent };