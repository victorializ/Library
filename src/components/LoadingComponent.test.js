import React from 'react';
import { shallow } from 'enzyme';

import { LoadingComponent } from './LoadingComponent';

describe('Loading component', () => {
    it('renders correctly when loading', () => {
        const error = {message: ''};
        expect(shallow(<LoadingComponent error={error} />)).toMatchSnapshot();
    });  
    it('renders correctly when error', () => {
        const error = {message: 'This is error'};
        expect(shallow(<LoadingComponent error={error} />)).toMatchSnapshot();
    });  
    it('renders correctly when data is loaded', () => {
        const error = {message: ''};
        expect(shallow(<LoadingComponent error={error} data='data'><p></p></LoadingComponent>)).toMatchSnapshot();
    });
}); 