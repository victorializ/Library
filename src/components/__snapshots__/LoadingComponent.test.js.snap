// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders correctly when data is loaded 1`] = `
<Fragment>
  <p />
  <Loader />
</Fragment>
`;

exports[`Loading component renders correctly when error 1`] = `
<Fragment>
  <ErrorMessage
    text="This is error"
  />
  <Loader />
</Fragment>
`;

exports[`Loading component renders correctly when loading 1`] = `
<Fragment>
  <Loader />
</Fragment>
`;
